### ISSUE TITLE
flox install / edit from within environment automatically rebuilds changes

### ISSUE THREAD

**Issue by ghudgins** [flox/flox#436]
# Acceptance Criteria

_install to current env inside of shell_
**Given** I have `init` and `flox activate` an environment
**When** I `flox install` a package
**Then** I see the environment refresh
**And** I can see the new package in my path

_uninstall from current env inside of shell_
**Given** I have `init` and `flox activate` an environment
**When** I `flox uninstall` a package
**Then** I see the environment refresh
**And** I can see the removed package is no longer in my path

_edit current env inside of shell_
**Given** I have `init` and `flox activate` an environment
**When** I `flox edit` a package (removing a package)
**Then** I see the environment refresh
**And** I can see the removed package is no longer in my path

_node: language specific variables are setup_
**Given** I have `init` and `flox activate` an environment
**When** I `flox install node` 
**Then** I see the environment refresh
**And** I can see node variables are set / hooks are re-run

_python: language specific variables are setup_
**Given** I have `init` and `flox activate` an environment
**When** I `flox install python3` 
**Then** I see the environment refresh
**And** I can see python3 variables are set / hooks are re-run

**Comment by ghudgins** [comment]
this looks resolved as of `c5d3dfb87f7cacc934ec505e7e93dfd6db19e746`

**Comment by mkenigs** [comment]
We're doing all the path and variables stuff, but we aren't re-running hooks.

The behavior of `pip` changes when you exit and re-activate:
```
> flox init
...
> SHELL=/bin/bash flox activate
...
flox [main] bash-3.2$ flox install pip
✅ 'pip' installed to environment main at /Users/matthew/flox/main/.flox
flox [main] bash-3.2$ pip install requests
error: externally-managed-environment

× This environment is externally managed
╰─> This command has been disabled as it tries to modify the immutable
    `/nix/store` filesystem.

    To use Python with Nix and nixpkgs, have a look at the online documentation:
    <https://nixos.org/manual/nixpkgs/stable/#python>.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.

[notice] A new release of pip is available: 23.0.1 -> 23.3.1
[notice] To update, run: python3.10 -m pip install --upgrade pip
flox [main] bash-3.2$ exit
> SHELL=/bin/bash flox activate
...
flox [main] bash-3.2$ pip install requests
ERROR: Could not find an activated virtualenv (required).
```

**Comment by ghudgins** [comment]
verified pip case
